### PR TITLE
Issue #2813 login to registry.redhat.io container registry

### LINKED ISSUE #2813: login to registry.redhat.io in case of using downstream bits.
We migrated from `registry.access.redhat.com` to `registry.redhat.io` which is behind the subscription and this need to be handled the same way as we handle the subscription manager bits.

### PR DESCRIPTION
**Pull request by praveenkumar**
Fixes #2813 

Steps to test the pull request

  1. MINISHIFT_ENABLE_EXPERIMENTAL=y  ./minishift start --iso-url <RHEL_iso> --no-provision
  2. ./minishift ssh
  3. ./docker pull registry.redhat.io/openshift3/ose-control-plane:latest => this should now able to pull the images.

### CODE REVIEW

File: pkg/minishift/registration/redhat.go [hunk 1]
```diff
@@ -90,10 +91,21 @@ func (registrator *RedHatRegistrator) Register(param *RegistrationParameters) er
 				param.Username,
 				minishiftStrings.EscapeSingleQuote(param.Password))
 
+			// prepare docker login command for registry.redhat.io
+			dockerLoginCommand := fmt.Sprintf("docker login --username %s --password %s %s",
+				param.Username,
+				minishiftStrings.EscapeSingleQuote(param.Password),
+				RedhatRegistry)
```
Comment (gbraad) [inline pkg/minishift/registration/redhat.go]: `RedhatRegistry` or `RedHatRegistry`... just like `RedHatRegistrator` and the official way of writing the name

**Comment by praveenkumar** [comment]
retest this please.

**Comment by praveenkumar** [comment]
retest this please.

**Comment by praveenkumar** [comment]
retest this please.

File: pkg/minishift/registration/redhat.go [hunk 2]
```diff
@@ -90,7 +89,28 @@ func (registrator *RedHatRegistrator) Register(param *RegistrationParameters) er
 				param.Username,
 				minishiftStrings.EscapeSingleQuote(param.Password))
 
-			fmt.Print("   Registration in progress ")
+			// prepare docker login command for registry.redhat.io
+			dockerLoginCommand := fmt.Sprintf("docker login --username %s --password %s %s",
```
Comment (budhram) [inline pkg/minishift/registration/redhat.go]: Not a blocker to merge this PR, but for readability or scalability, this string can be extracted as: 
```
dockerLoginTmpl = "docker login --username %s --password %s %s"
dockerLoginCommand := fmt.Sprintf(dockerLoginTmpl, param.Username, ......)
```

**Pull request merged**